novawallet/Modules/Vote/Governance/ViewModel/ReferendumDisplayStringFactory.swift

```diff
@@ -12,13 +12,135 @@ protocol ReferendumDisplayStringFactoryProtocol {
         chain: ChainModel,
         locale: Locale
     ) -> String?
+
+    func createReferendumVotes(
+        from referendum: ReferendumLocal,
+        offchainVotingAmount: ReferendumVotingAmount?,
+        chain: ChainModel,
+        locale: Locale
+    ) -> ReferendumVotesViewModel
+
+    func createDirectVotesViewModel(
+        from vote: ReferendumAccountVoteLocal,
+        chain: ChainModel,
+        locale: Locale
+    ) -> [YourVoteRow.Model]
+
+    func createDelegatorVotesViaDelegateViewModel(
+        from vote: GovernanceOffchainVoting.DelegateVote,
+        delegateName: String?,
+        chain: ChainModel,
+        locale: Locale
+    ) -> [YourVoteRow.Model]
+}
+
+private typealias ReferendumDisplayStringFactory = ReferendumDisplayStringFactoryProtocol & BaseReferendumDisplayStringFactory
+
+enum ReferendumDisplayStringFactoryProvider {
+    static func factory(
+        for govType: GovernanceType,
+        formatterFactory: AssetBalanceFormatterFactoryProtocol = AssetBalanceFormatterFactory()
+    ) -> ReferendumDisplayStringFactoryProtocol {
+        if govType == .governanceV2 {
+            OpenGovReferendumDisplayStringFactory(formatterFactory: formatterFactory)
+        } else {
+            Gov1ReferendumDisplayStringFactory(formatterFactory: formatterFactory)
+        }
+    }
+}
+
+// MARK: OpenGov Factory
+
+final class OpenGovReferendumDisplayStringFactory: ReferendumDisplayStringFactory {
+    let formatterFactory: AssetBalanceFormatterFactoryProtocol
+
+    init(formatterFactory: AssetBalanceFormatterFactoryProtocol = AssetBalanceFormatterFactory()) {
+        self.formatterFactory = formatterFactory
+    }
+
+    func createViewModel(
+        title: String,
+        value: BigUInt?,
+        chain: ChainModel,
+        locale: Locale
+    ) -> VoteRowView.Model? {
+        let loadableValueString: LoadableViewModelState<String>
+
+        if
+            let value,
+            let valueString = createVotes(
+                from: value,
+                chain: chain,
+                locale: locale
+            ) {
+            loadableValueString = .loaded(value: valueString)
```

Comment:
I don't like that approach. The factory intended to be one that returns formatted string. It shouldn't decide whether to ask for loading - it obviously don't have enough information. For abstain we already have supportsAbstain flag passed to the presenter so we can decide based on this flag what to display. For displaying off chain votes we also can introduce ```supportsOffchainVotes``` flag to make the behaviour simpler and obvious.

Reply:
fixed